functions/utils/cache.ts

```diff
@@ -0,0 +1,25 @@
+class Cache {
+  async match(request: string) {
```

Comment:
You could make this its own type:

```
interface Data {
  title: string
  image: string
  url: string
}
```

```suggestion
  async match(request: string): { title: string; image: string; url: string } {
```